builder-base/scripts/download_golang.sh

```diff
@@ -33,19 +33,41 @@ function build::go::download(){
     local outputDir=${2}
     local arch=${3}
 
+    mkdir $outputDir/tmp
+
     for artifact in golang golang-bin; do
         local filename="$outputDir/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm"
         if [ ! -f $filename ]; then
             curl -sSL --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm -o $filename
         fi
+
+        local shaCheckFilename="$outputDir/tmp/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm"
+        curl -sSL --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm -o $shaCheckFilename
+        curl -sSL --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/$arch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.rpm.sha256 -o $shaCheckFilename.sha256
+
+        if [[ $(sha256sum ${shaCheckFilename} | cut -d' ' -f1) != $(cut -d' ' -f1 "${shaCheckFilename}.sha256") ]] ; then 
+            echo "Checksum doesn't match!"
+            exit 1
+        fi
     done
 
     for artifact in golang-docs golang-misc golang-tests golang-src; do
         local filename="$outputDir/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm"
         if [ ! -f $filename ]; then
             curl -sSL --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $filename
         fi
+
+        local shaCheckFilename="$outputDir/tmp/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.$arch.noarch.rpm"
+        curl -sSL --retry 5 https://distro.eks.amazonaws.com/golang-go$version/releases/$RELEASE_NUMBER/$arch/RPMS/noarch/$artifact-$version-$RELEASE_NUMBER.amzn2.eks.noarch.rpm -o $shaCheckFilename
```

Comment:
so, we don't want to perform the checksum check if the above check takes place-- the `! -f` indicates that if that file exists, we skip the download; we do this so that we can over-ride the golang RPM used with a local one. In this case, we don't want to perform the checksum check, as we're never using the downloaded RPM, but just using a local one.